### PR TITLE
add support for pagination for non-nested dynamic groups

### PR DESCRIPTION
**Pull request by sashankaryal**
## What changes are proposed in this pull request?

Add support for pagination UI for non-nested dynamic groups.

Note: flashlight offers a sequential "infinite scroll" UI, whereas the pagination bar offers a random access UI. Reconciling the two into a common interface is non-trivial, and so this PR takes the easy route of only offering one UI at a time. Sequential UI (carousel / flashlight) is the default.

Todo: e2e tests

Screenshot 2023-09-28 at 11 48 59 PM

## How is this patch tested? If it is not, please explain why.
All E2E green.

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

**Review by benjaminpkane — COMMENTED** [review]
Clean stuff!! I think some app unit tests are failing. I spent a few minutes trying to fix, but I think we are running into circularity issues in `@fiftyone/state`

File: app/packages/core/src/components/Actions/GroupMediaVisibilityContainer.tsx [hunk 1]
```diff
@@ -38,36 +38,67 @@ const GroupMediaVisibilityPopout = ({
...
+    );
+
+    return toReturn;
+  }, [
+    pointCloudSliceExists,
+    isSequentialAccessAllowed,
+    isCarouselVisible,
+    isMainVisible,
+    isSlotVisible,
+  ]);
```
[the remark this answers is not in the document]
Comment (sashankaryal) [reply app/packages/core/src/components/Actions/GroupMediaVisibilityContainer.tsx:91]: No straightforward way, but perhaps with a custom rule!

**Comment by sashankaryal** [comment]
> Clean stuff!! I think some app unit tests are failing. I spent a few minutes trying to fix, but I think we are running into circularity issues in `@fiftyone/state`

Thanks for taking a look!
I moved the `getBrowserStorageEffectForKey()` effect into a different module. That should solve the circular import issue.

**Review by benjaminpkane — APPROVED** [review]
LGTM. I think e2e would be valuable, but it doesn't need to be extensive, e.g. just asserting that the toggling modes hides/shows the carousel/pager

**Pull request merged**